### PR TITLE
Fix: Thread running at User-initiated quality-of-service for Session replay

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

AVFoundation video maker runs on a lower QoS than user-initiated processes, and the thread sanitizer complains about waiting for it on a higher-level thread. We are changing the QoS of our call to create video.

## :bulb: Motivation and Context

Reported in here: https://github.com/getsentry/sentry/discussions/74322#discussioncomment-10934816

## :green_heart: How did you test it?

Unit test

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x]  No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ via the option `swizzleClassNameExclude`.
 - Finish TTID correctly when viewWillAppear is skipped (#4417)
 - Swizzling RootUIViewController if ignored by `swizzleClassNameExclude` (#4407)
 - Data race in SentrySwizzleInfo.originalCalled (#4434)
+- Thread running at User-initiated quality-of-service for session replay (#4439)
```
Comment (philipphofmann) [inline CHANGELOG.md:22]: `l`:
```suggestion
- Thread running at user-initiated quality-of-service for session replay (#4439)
```

File: Sources/Sentry/SentrySessionReplayIntegration.m [hunk 2]
```diff
@@ -281,6 +281,10 @@ - (void)startWithOptions:(SentryReplayOptions *)replayOptions
         = (NSInteger)(shouldReplayFullSession ? replayOptions.sessionSegmentDuration + 1
                                               : replayOptions.errorReplayDuration + 1);
 
+    
+    dispatch_queue_attr_t attributes = dispatch_queue_attr_make_with_qos_class(                                                                               DISPATCH_QUEUE_SERIAL, DISPATCH_QUEUE_PRIORITY_LOW, 0);
+    SentryDispatchQueueWrapper* dispatchQueue = [[SentryDispatchQueueWrapper alloc] initWithName:"Sentry Session Replay" attributes:attributes];
```
Comment (philipphofmann) [inline Sources/Sentry/SentrySessionReplayIntegration.m:286]: `m`: We should align the dispatch queue names with the Apple naming scheme for queues. I'm going to open a PR to rename the other queues.

CleanShot 2024-10-15 at 13 47 01@2x


```suggestion
    SentryDispatchQueueWrapper* dispatchQueue = [[SentryDispatchQueueWrapper alloc] initWithName:"io.sentry.session-replay" attributes:attributes];
```

**Review by philipphofmann — APPROVED** [review]
LGTM, thanks.

File: Sources/Sentry/SentrySessionReplayIntegration.m (hunk 2 above)
Comment (philipphofmann) [reply Sources/Sentry/SentrySessionReplayIntegration.m:286]: PR to algin the dispatch queue names: https://github.com/getsentry/sentry-cocoa/pull/4442

Comment (brustolin) [reply Sources/Sentry/SentrySessionReplayIntegration.m]: Makes sense

File: Sources/Sentry/SentrySessionReplayIntegration.m [hunk 3]
```diff
@@ -281,6 +281,12 @@ - (void)startWithOptions:(SentryReplayOptions *)replayOptions
         = (NSInteger)(shouldReplayFullSession ? replayOptions.sessionSegmentDuration + 1
                                               : replayOptions.errorReplayDuration + 1);
 
+    dispatch_queue_attr_t attributes = dispatch_queue_attr_make_with_qos_class(
+        DISPATCH_QUEUE_SERIAL, DISPATCH_QUEUE_PRIORITY_LOW, 0);
+    SentryDispatchQueueWrapper *dispatchQueue =
+        [[SentryDispatchQueueWrapper alloc] initWithName:"Sentry Session Replay"
```
Comment (brustolin) [inline Sources/Sentry/SentrySessionReplayIntegration.m:287]:
```suggestion
        [[SentryDispatchQueueWrapper alloc] initWithName:"io.sentry.session-replay"
```

**Pull request merged**